File: src/main/kotlin/assertions/UiAssert.java
package assertions;

import com.microsoft.playwright.Page;
import org.assertj.core.api.Assertions;

public final class UiAssert {
    private UiAssert() {}

    public static void urlContains(Page page, String part) {
        Assertions.assertThat(page.url()).contains(part);
    }

    public static void titleIs(Page page, String expected) {
        Assertions.assertThat(page.title()).isEqualTo(expected);
    }
}

File: src/main/kotlin/assertions/UiAssert.kt
package assertions

import com.microsoft.playwright.Page
import org.assertj.core.api.Assertions

object UiAssert {
    fun urlContains(page: Page, part: String) {
        Assertions.assertThat(page.url()).contains(part)
    }

    fun titleIs(page: Page, expected: String) {
        Assertions.assertThat(page.title()).isEqualTo(expected)
    }

    fun assertVisible(page: Page, locator: String) {
        Assertions.assertThat(page.isVisible(locator))
            .withFailMessage("Expected element '$locator' to be visible")
            .isTrue()
    }

    fun assertText(page: Page, locator: String, expected: String) {
        val actual = page.textContent(locator) ?: ""
        Assertions.assertThat(actual)
            .withFailMessage("Text for '$locator' did not match")
            .isEqualTo(expected)
    }

    fun assertContainsText(page: Page, locator: String, part: String) {
        val actual = page.textContent(locator) ?: ""
        Assertions.assertThat(actual)
            .withFailMessage("Text for '$locator' did not contain expected part")
            .contains(part)
    }
}